chapters/scoping.tex

```diff
@@ -98,7 +98,8 @@ \subsection{Composite Name Lookup}\label{composite-name-lookup}
 \item
   If the identifier denotes a class, that class is temporarily flattened (as if instantiating a component without modifiers of this class, see \cref{modification-environment}) and using the enclosing classes of the denoted class.
   The rest of the name (e.g., \lstinline!B! or \lstinline!B.C!) is looked up among the declared named elements of the temporary flattened class.
-  If the class does not satisfy the requirements for a package, the lookup is restricted to encapsulated\indexinline{encapsulated} elements only.
+  If the class does not satisfy the requirements for a \lstinline!package!, the lookup is restricted to encapsulated\indexinline{encapsulated} elements only.
+  The special case where the class satisfies the requirements for a \lstinline!package!, but is not declared as \lstinline!package! or \lstinline!operator!, is deprecated.
```

Comment:
Two things:
1. It's not ideal that the non-deprecated feature is expressed using a sentence about deprecation.
2. Isn't the concept referred to as `operator record` instead of just `operator`?

Put more constructively:
```suggestion
  Unless the class is a \lstinline!package! or \lstinline!operator record!, the lookup is restricted to encapsulated\indexinline{encapsulated} elements only.
  A deprecated semantics is that the lookup restriction does not apply to any class that satisfies the requirements for a \lstinline!package!.
```

Reply:
Good, except for the 'operator record' part.

There are three different kinds of specialized classes with `operator` in their name: `operator record`, `operator`, and `operator function`, and it is `operator` that is package-like.

As an example:
```
operator record Complex
   Real re,im;
   encapsulated operator '-'
       function negate ... end negate;
       function subtract ... end subtract;
  end '-';
  encapsulated operator function '+' end '+';
end Complex;
```
I can add the rest and an explanation for this.

Reply:
Have added it.

Reply:
Looks OK.  Resolving.